src/include/clSPARSE.h

```diff
@@ -360,6 +360,40 @@ clsparseDCsrMatrixfromFile( clsparseCsrMatrix* csrMatx, const char* filePath, cl
         const cldenseVector* y,
         const clsparseControl control );
 
+    // functions for tests of scan transformation, as simple as possible
```

Comment:
Interesting, i had not thought about testing internal algorithms.  In such cases, I believe it would be valid to create a new test executable that leverages our private global header file: clSPARSE-private.hpp.